src/database/DataTypes/Stocktake.js

```diff
@@ -198,6 +199,29 @@ export class Stocktake extends Realm.Object {
     return getTotal(this.items, 'numberOfBatches');
   }
 
+  /**
+   * Get if stocktake has any item without reason set
+   *
+   * @return {boolean}
+   */
+  get hasReasonNotSet() {
+    return this.items.some(item => !item.hasReasonSet === false);
+  }
+
+  /**
+   * Check wether Reason i.e. Option is activated in mSupply desktop
+   * Return True if all positive and negative reasons are activated
+   *
+   * @return {boolean}
+   */
+  // eslint-disable-next-line class-methods-use-this
+  get isReasonActive() {
```

Comment:
@arjunSussol @sworup: Why do we need to check items without reason if the no reason option is activated? I think it can make a mobile slow to go through every item if the reason option is not activated.
I think it better if we can check whether the reason was activated or not first then if yes then only look for items in stock take. What do you suggest?

Reply:
Good suggestion @anildahalsussol

Reply:
> @arjunSussol @sworup: Why do we need to check items without reason if the no reason option is activated? I think it can make a mobile slow to go through every item if the reason option is not activated. I think it better if we can check whether the reason was activated or not first then if yes then only look for items in stock take. What do you suggest?

Thanks @anildahalsussol and @sworup 
Fixed by https://github.com/openmsupply/mobile/pull/5029/commits/dd026a2196593c848db885519794fb5e5bd4894a